src/static/helpers/customApi.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {BodyInit} from 'node-fetch';
+import {PathParameters} from './types';
+import {doFetch} from './fetchHelper';
+import TemplateURL from '../templateUrl';
+import {ClientConfigInit} from '../clientConfig';
+
+// Helper method to find Content Type header
+// returns true if it exists, false otherwise
+const contentTypeHeaderExists = (
+  headers: Record<string, string> | undefined
+) => {
+  let foundHeader = false;
+  if (headers) {
+    foundHeader = Boolean(
+      Object.keys(headers).find(key => key.toLowerCase() === 'content-type')
+    );
+  }
+  return foundHeader;
+};
+
+export interface CustomParams {
+  apiName?: string;
+  apiVersion?: string;
+  endpointPath?: string;
+  organizationId?: string;
+  shortCode: string;
+  [key: string]: unknown;
+}
```

Comment:
Is it possible to extend from any existing types? I'm assuming there is a type that includes the org id and short code, etc. If that type has to change, we would have to change it in 2 places now.

Reply:
Additionally, why are the params optional?

Reply:
The only existing interface is `BaseUriParameters`:

```
export interface BaseUriParameters {
  shortCode: string;
  version?: string; // Optional, will default to "v1" if not provided.
}
```

and we're required to extend it when passing in a type for client config:

```
/**
 * Base options that can be passed to the `ClientConfig` class.
 */
export interface ClientConfigInit<Params extends BaseUriParameters> {
  ...
  parameters: Params;
}
```

Reply:
Params are optional as they can be passed in through `clientConfig.parameters` or `options.parameters`. We'll throw an error if required parameters are missing